src/MeasurementCodeInjector.php

```diff
@@ -37,11 +55,90 @@ public function setActivePlugins() {
         $this->activePlugins = $this->pluginDetector->getActivePlugins();
     }
 
-    public function printActivePlugins() {
-        foreach($this->activePlugins as $activePlugin){
-            echo $activePlugin . '<br>';
+    /**
+     * Creates list of measurement event configurations and javascript to inject
+     */
+    public function injectEventTracking() {
+        foreach($this->activePlugins as $pluginName) {
+            $measurementEventList = $this->eventFactory->createMeasurementEventList($pluginName);
+            if($measurementEventList != null) {
+                foreach ($measurementEventList->getEvents() as $measurementEvent) {
+                    $this->configureEvent($measurementEvent);
+                }
+            }
         }
+        ?>
+        <script>
+            /**
+             * Keeps track of Ninja forms that have already been tracked
+             *
+             * @type Array of elements
+             */
+            let ninjaFormsAddedNodes = [];
+
+            /**
+             * Adds event listener to element that corresponds to the event
+             *
+             * @param config - configuration for a trackable event
+             */
+            function addListeners(config) {
+                let nodeList = document.querySelectorAll(config.selector);
+                for(node of nodeList) {
+                    if(!ninjaFormsAddedNodes.includes(node)) {
+                        node.addEventListener(config.on, function () {
+                            alert('Got an event called: '.concat(config.action));
+                        });
+                        if(config.pluginName == 'Ninja Forms') ninjaFormsAddedNodes.push(node);
+                    }
+                }
+            }
+
+            /**
+             * Adds first or second layer event listeners on DOM loaded
+             */
+            jQuery(function($){
+                let eventConfigurations = <?php echo json_encode($this->eventConfigurations); ?>;
+                for(config of eventConfigurations) {
+                    if(config.secondLayerOn === null) {
+                        addListeners(config);
+                    }else{
+                        let secondLayerNode;
+                        switch(config.secondLayerSelector) {
+                            case 'document':
+                                secondLayerNode = $(document);
+                                break;
+                            case 'document.body':
+                                secondLayerNode = $(document.body);
+                                break;
+                            default:
+                                secondLayerNode = $(config.secondLayerSelector);
+                                break;
+                        }
+                        const configCopy = JSON.parse(JSON.stringify(config));
+                        secondLayerNode.on(config.secondLayerOn, function(){
```

Comment:
Does not need to be addressed in this PR. Can we explore handling dynamically added elements without relying on an use case specific event to attach the listeners.